jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -263,7 +285,15 @@ private SpanContext createNewContext(String debugId) {
         }
       }
 
-      return new SpanContext(id, id, 0, flags);
+      // Check if active span should be established as CHILD_OF relationship
+      if (!ignoreActiveSpan && null != activeSpanSource.activeSpan()) {
+        SpanContext parentContext = (SpanContext) activeSpanSource.activeSpan().context();
+        asChildOf(parentContext);
+        traceId = parentContext.getTraceId();
+        parentId = parentContext.getSpanId();
+      }
+
+      return new SpanContext(traceId, id, parentId, flags);
```

Comment:
@yurishkuro Update applied - although not sure it is right, as the conditions for adding the childOf for the active span is that no other explicit references have been added? Possibly it just needs an extract condition - e.g. `references.isEmpty() && !ignoreActiveSpan && ...` ?